src/vs/server/webClientServer.ts

```diff
@@ -295,7 +295,12 @@ export class WebClientServer {
 			id: generateUuid(),
 			providerId: 'github',
 			accessToken: this._environmentService.args['github-auth'],
-			scopes: [['user:email'], ['repo']]
+			/**
+			 * Add a set of scopes for the pull request extension which also wants
+			 * read:user.
+			 * @author coder
+			 */
+			scopes: [['read:user', 'user:email', 'repo'], ['user:email'], ['repo']]
```

Comment:
❓ How come they have to be duplicated like this? I would think it would be enough to do:
```suggestion
			scopes: [['read:user'], ['user:email'], ['repo']]
```

Reply:
Yup, I thought so too but the scopes have to match exactly.  So if an extension wants `['repo']` but the token offers both `['repo', 'read:user']` then it will not match.

You can see here they do a strict equals on the scopes array:

https://github.com/coder/vscode/blob/676edaabfd8a9b81d274dab005d38df7c387a0f8/extensions/github-authentication/src/github.ts#L73-L75

It does feel like a bug so maybe we should patch this instead but idk.

Reply:
AH...that make sense. Thanks for explaining. When we get to patches, I will see if it's a bug and we can raise upstream.